src/NewsService.Mappers/Responses/NewsResponseMapper.cs

```diff
@@ -1,110 +1,49 @@
-        var response = _userRequestClient.GetResponse<IOperationResult<IGetUserDataResponse>>(request).Result;
-
-        if (!response.Message.IsSuccess)
-        {
-          _logger.LogWarning($"Can't found user. Reason: '{string.Join(',', response.Message.Errors)}'");
-        }
-        else
-        {
-          fullName = $"{response.Message.Body.FirstName} {response.Message.Body.LastName} {response.Message.Body.MiddleName}".Trim();
-        }
-      }
-      catch (Exception exception)
-      {
-        _logger.LogError(exception, "Exception on get user data request.");
-      }
-
-      return fullName;
-    }
-
-    private string GetDepartmentName(Guid departmentId)
-    {
-      string name = null;
-
-      try
-      {
-        var request = IGetDepartmentRequest.CreateObj(null, departmentId);
-        var response = _departmentRequestClient.GetResponse<IOperationResult<IGetDepartmentResponse>>(request).Result;
-
-        if (!response.Message.IsSuccess)
-        {
-          _logger.LogWarning($"Can't found department name. Reason: '{string.Join(',', response.Message.Errors)}'");
-        }
-        else
-        {
-          name = response.Message.Body.Name;
-        }
-      }
-      catch (Exception exception)
-      {
-        _logger.LogError(exception, "Exception on get department data request.");
-      }
-
-      return name;
+      _httpContextAccessor = httpContextAccessor;
+      _userInfoMapper = userInfoMapper;
     }
 
-    public NewsResponseMapper(
-      IRequestClient<IGetUserDataRequest> userClient,
-      IRequestClient<IGetDepartmentRequest> departmentClient,
-      ILogger<NewsResponseMapper> logger)
-    {
-      _userRequestClient = userClient;
-      _departmentRequestClient = departmentClient;
-      _logger = logger;
-    }
-
-    public NewsResponse Map(DbNews dbNews)
+    public NewsResponse Map(DbNews dbNews, DepartmentInfo department, UserData author)
     {
       if (dbNews == null)
       {
         return null;
       }
 
-      Department department = null;
-      if (dbNews.DepartmentId != null && dbNews.DepartmentId != Guid.Empty)
-      {
-        department = new Department { Id = (Guid)dbNews.DepartmentId, Name = GetDepartmentName((Guid)dbNews.DepartmentId) };
-      }
+      List<ImageInfo> avatarImage = new();
 
       return new NewsResponse
       {
         Id = dbNews.Id,
         Preview = dbNews.Preview,
         Content = dbNews.Content,
         Subject = dbNews.Subject,
-        Author = new User { Id = dbNews.AuthorId, FullName = GetUserFullName(dbNews.AuthorId) },
+        Pseudonym = dbNews.Pseudonym,
+        Author = _userInfoMapper.Map(author, avatarImage?.FirstOrDefault(ai => ai.Id == author.ImageId)),
         Department = department,
         IsActive = dbNews.IsActive,
-        CreatedAtUtc = DateTime.UtcNow
+        CreatedAtUtc = DateTime.UtcNow,
+        CreatedBy = _httpContextAccessor.HttpContext.GetUserId()
```

Comment:
What is this? it's response mapper, you must set values from db

Reply:
done